src/fuzzer_client.c

```diff
@@ -0,0 +1,814 @@
+               thread_id, (unsigned long)pthread_self());
+
+    /* Set global abort flag to stop all threads */
+    global_abort_all_threads = 1;
+
+    incrCounter(COUNTER_MALFORMED_REPLIES);
+    sdsfree(cmd_str);
+}
+
+static void handleCommandTimeout(Command *cmd, time_t start_time) {
+    /* Timeout error */
+    time_t current_time = time(NULL);
+    double elapsed_time = difftime(current_time, start_time);
+
+    char cmd_buffer[256];
+    formatCommandToBuffer(cmd, cmd_buffer, sizeof(cmd_buffer));
+    logMessage(LOG_ERROR, "[Thread %d] TIMEOUT: Command timed out after %.1f seconds (max %d seconds) - Command: %s",
+               thread_id, elapsed_time, COMMAND_TIMEOUT_SEC, cmd_buffer);
+    logMessage(LOG_ERROR, "[Thread %d] ABORTING THREAD due to timeout (pthread_id: %lu)",
+               thread_id, (unsigned long)pthread_self());
+}
+
+/* Helper function to handle command errors and update counters */
+static CommandResult handleCommandError(valkeyContext *ctx, Command *cmd, time_t start_time) {
+    CommandResult result = CMD_ERROR;
+
+    if (ctx->err == VALKEY_ERR_IO && (errno == EAGAIN || errno == EWOULDBLOCK || errno == ETIMEDOUT)) {
+        handleCommandTimeout(cmd, start_time);
+        result = CMD_TIMEOUT;
+    } else if (isServerDisconnected(ctx)) {
+        /* Connection closed */
+        logMessage(LOG_DEBUG, "Connection closed by server: %s", ctx->errstr);
+        result = CMD_DISCONNECTED;
+    } else if (ctx->err == VALKEY_ERR_PROTOCOL) {
+        handleMalformedReply(ctx, cmd);
+    } else {
+        sds cmd_str = formatCommandString(cmd);
+        char error_msg[512];
+        snprintf(error_msg, sizeof(error_msg), "Error getting reply: %s (errno: %d)",
+                 ctx->errstr, errno);
+        addError(thread_error_list, error_msg, cmd_str);
+        incrCounter(COUNTER_TOTAL_ERRORS);
+        sdsfree(cmd_str);
+    }
+
+    return result;
+}
+
+/* Helper function to log reply directly without SDS allocation */
+static void logReplyDebug(const char *command, valkeyReply *reply) {
+    if (current_log_level < LOG_DEBUG) return;
+
+    switch (reply->type) {
+    case VALKEY_REPLY_STRING:
+    case VALKEY_REPLY_STATUS:
+    case VALKEY_REPLY_ERROR:
+        if (reply->str) {
+            logMessage(LOG_DEBUG, "Command -> Reply: %s -> %.100s", command, reply->str);
+        } else {
+            logMessage(LOG_DEBUG, "Command -> Reply: %s -> (empty)", command);
+        }
+        break;
+    case VALKEY_REPLY_ARRAY:
+        logMessage(LOG_DEBUG, "Command -> Reply: %s -> (array of %zu elements)", command, reply->elements);
+        break;
+    case VALKEY_REPLY_INTEGER:
+        logMessage(LOG_DEBUG, "Command -> Reply: %s -> %lld", command, reply->integer);
+        break;
+    case VALKEY_REPLY_NIL:
+        logMessage(LOG_DEBUG, "Command -> Reply: %s -> (nil)", command);
+        break;
+    default:
+        logMessage(LOG_DEBUG, "Command -> Reply: %s -> (unknown reply type)", command);
```

Comment:
nit: `Command -> Reply` feels redundant, we could remove it.

Reply:
Thanks, Done.